pvlib/mlfm.py

```diff
@@ -0,0 +1,804 @@
+    c_5 : float, default 0
+        Coefficient for wind speed dependence.
+    c_6 : float, default 0
+        Coefficient for dependence on inverse irradiance.
+
+    Returns
+    -------
+    mlfm_6 : Series
+        Predicted values.
+
+    References
+    ----------
+    .. [1] Steve Ransome (SRCL) and Juergen Sutterlueti (Gantner Instruments)
+       "Quantifying Long Term PV Performance and Degradation under Real Outdoor
+       and IEC 61853 Test Conditions Using High Quality Module IV Measurements"
+       36th EU PVSEC, Marseille, France. September 2019
+     '''
+    mlfm_out = c_1 + c_2 * (dmeas['temp_module'] - T_STC) + \
+        c_3 * np.log10(dmeas['poa_global'] / G_STC) + \
+        c_4 * (dmeas['poa_global'] / G_STC) + \
+        c_6 / (dmeas['poa_global'] / G_STC)
+    if 'wind_speed' in dmeas.columns:
+        mlfm_out += c_5 * dmeas['wind_speed']
+    return mlfm_out
+
+
+def mlfm_fit(data, var_to_fit):
+    '''
+    Fit MLFM to data.
+
+    Parameters
+    ----------
+    data : DataFrame
+        Must include columns:
+
+        * 'poa_global' global plane of array irradiance. [W/m^2]
+        * 'temp_module' module temperature. [C]
+
+        Must include column named ``var_to_fit``.
+
+        May include optional column:
+
+            * 'wind_speed' wind speed [m/s].
+
+    var_to_fit : string
+        Column name in ``data`` containing variable being fit.
+
+    Returns
+    -------
+    pred : Series
+        Values predicted by the fitted model.
+
+    coeff : list
+        Model coefficients ``c_1`` to ``c_6``.
+
+    resid : Series
+        Residuals of the fitted model.
+
+    coeff_err : list
+        Standard deviation of error in each model coefficient.
+
+    See also
+    --------
+    mlfm_6
+    '''
+
+    # drop missing data
+    data = data.dropna()
+
+    c5_zero = 'wind_speed' not in data.columns
+    # if wind_speed is not present, add it
+    if c5_zero:
+        data['wind_speed'] = 0.
+
+    # define function name
+    func = mlfm_6
+
+    # setup initial values and initial boundary conditions
+
+    # initial   c1    c2    c3    c4    c5   c6<0
+    p_0 = (1.0, 0.01, 0.01, 0.01, 0.01, -0.01)
+    # boundaries
+    bounds = ([ -2,   -2,   -2,   -2,   -2,    -2],
+              [  2,    2,    2,    2,    2,     0])
+
+    coeff, pcov = optimize.curve_fit(
+        f=func,                 # fit function
+        xdata=data,            # input data
+        ydata=data[var_to_fit],  # fit parameter
+        p0=p_0,                 # initial
+        bounds=bounds           # boundaries
+    )
+
+    # if data has no wind_speed measurements then c_5 coefficient is
+    # meaningless but a non-zero value may have been returned.
+    if c5_zero:
+        coeff[4] = 0.
+
+    # get error of mlfm coefficients as sqrt of covariance
+    perr = np.sqrt(np.diag(pcov))
+    coeff_err = list(perr)
+
+    # save fit and error to dataframe
+    pred = mlfm_6(data, coeff[0], coeff[1], coeff[2], coeff[3], coeff[4],
```

Comment:
consider using *coeff

Reply:
Thanks

Reply:
I think that you can pass `full_output=True` to `scipy.optimize.curve_fit`, then get the `pred` variable pre-calculated as `infodict["fvec"]` from the additional output `infodict` from `scipy.optimize.curve_fit`.

`full_output=True` also provides two more outputs: `mesg` and `ierint`, which you should inspect for convergence information, possibly warning the user and/or passing back to the caller.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html